src/codegen.cpp

```diff
@@ -3331,12 +3331,19 @@ static LLVMValueRef ir_render_bit_cast(CodeGen *g, IrExecutableGen *executable,
             LLVMPointerType(get_llvm_type(g, wanted_type), 0) : get_llvm_type(g, wanted_type);
         return LLVMBuildBitCast(g->builder, value, wanted_type_ref, "");
     } else if (actual_is_ptr) {
+        // A scalar is wanted but we got a pointer
         LLVMTypeRef wanted_ptr_type_ref = LLVMPointerType(get_llvm_type(g, wanted_type), 0);
         LLVMValueRef bitcasted_ptr = LLVMBuildBitCast(g->builder, value, wanted_ptr_type_ref, "");
         uint32_t alignment = get_abi_alignment(g, actual_type);
         return gen_load_untyped(g, bitcasted_ptr, alignment, false, "");
     } else {
-        zig_unreachable();
+        // A pointer is wanted but we got a scalar
+        // Spill the scalar into a local memory location and take its address
+        LLVMValueRef spill = build_alloca(g, actual_type, "bc", 0);
```

Comment:
All alloca instructions have to be done up front in the entry block, no allocas allowed mid-function. You'll have to detect this situation when emitting a bit cast instruction in ir.cpp and set it up. You can see an example of this in `ir_analyze_err_wrap_payload`.

Reply:
Yeah, I didn't take into account the async machinery. I'll amend the patch asap.

Reply:
I hadn't even considered async, that's a good point. Even for non-async, putting all the allocas up front generates better code because the stack pointer decrements only once at the beginning of the function, and functions have more predictable stack size (once we start taking advantage of llvm emitting stack size annotations)